docs/Protected_build_porting_guide.md

```diff
@@ -0,0 +1,363 @@
+# Protected Build Porting Guide
+
+
+## Contents
+- [Inroduction](#introduction)
+	- [Assumptions](#assumptions)
+	- [What is Protected Build](#what-is-protected-build)
+- [Steps for porting protected build in a new chipset](#Steps-for-porting-protected-build-in-a-new-chipset)
+	- [Creating new configuration files](#Creating-new-configuration-files)
+	- [Creating linker scripts](#Creating-linker-scripts)
+	- [Userspace initialization](#Userspace-initialization)
+	- [Mpu initialization](#Mpu-initialization)
+	- [Allocation of kernel and user heaps](#Allocation-of-kernel-and-user-heaps)
+	- [Changes in make.defs and makefile](#Changes-in-make.defs-and-makefile)
+	- [Changes in the download method](#Changes-in-the-download-method)
+
+
+## Introduction
+This is a guide for porting Protected Build for a new chipset.
+
+### Assumptions
+- TizenRT 3.0 flat build has been successfully ported and tested on the target CHIPSET board
+- Generic code relating to TizenRT protected build is already present in the codebase (MACRO: CONFIG_ BUILD_PROTECTED)
+- In order to bring up protected build, changes are to be made specific to the board
+- Information is available about the memory map of the board, partition information and memory organization
+
+### What is Protected Build
+- In protected build, two separate binaries namely kernel binary and user binary are created.
```

Comment:
kernel space binary & user space binary.  There can be multiple binaries in each space.

Reply:
Done